modules/sdk-core/src/bitgo/utils/tss/baseTypes.ts

```diff
@@ -0,0 +1,146 @@
+import { SerializedKeyPair } from 'openpgp';
+import { IRequestTracer } from '../../../api';
+import { KeychainsTriplet } from '../../baseCoin';
+import { Keychain } from '../../keychain';
+import { Memo, WalletType } from '../../wallet/iWallet';
+import { EDDSA } from '../../../account-lib/mpc/tss';
+
+export type TxRequestVersion = 'full' | 'lite';
+
+export interface PrebuildTransactionWithIntentOptions {
+  reqId: IRequestTracer;
+  intentType: string;
+  sequenceId?: string;
+  recipients: {
+    address: string;
+    amount: string | number;
+    tokenName?: string;
+  }[];
+  comment?: string;
+  memo?: Memo;
+  tokenName?: string;
+  nonce?: string;
+}
+
+export type TxRequestState =
+  | 'pendingApproval'
+  | 'canceled'
+  | 'rejected'
+  | 'initialized'
+  | 'pendingDelivery'
+  | 'delivered'
+  | 'pendingUserSignature'
+  | 'signed';
+
+export type TransactionState =
+  | 'initialized'
+  | 'pendingSignature'
+  | 'signed'
+  | 'held'
+  | 'delivered'
+  | 'invalidSignature'
+  | 'rejected';
+
+// Type used to sign a TSS transaction
+export type SignableTransaction = {
+  // unsigned transaction in broadcast format
+  serializedTxHex: string;
+  // portion of a transaction used to generate a signature
+  signableHex: string;
+};
+
+export type UnsignedTransactionTss = SignableTransaction & {
+  // derivation path of the signer
+  derivationPath: string;
+  // transaction fees
+  feeInfo?: {
+    fee: number;
+    feeString: string;
+  };
+  coinSpecific?: Record<string, unknown>;
+  parsedTx?: unknown;
+};
+
+export type TxRequest = {
+  txRequestId: string;
+  walletId: string;
+  walletType: WalletType;
+  version: number;
+  enterpriseId?: string;
+  state: TxRequestState;
+  date: string;
+  userId: string;
+  intent: unknown; // Should override with sig scheme specific intent
```

Comment:
More safe to use `unknown` as it will prompt people to do proper typechecing before use